### PR TITLE
Optimized what is cached in local variables

### PR DESCRIPTION
**Pull request by Brechtpd**
This saves 1500 gas. To be honest I'm not sure why this saves so much gas.

### CODE REVIEW

File: contracts/LoopringProtocolImpl.sol [hunk 1]
```diff
@@ -497,32 +497,32 @@ contract LoopringProtocolImpl is LoopringProtocol {
         uint p = 0;
         for (uint i = 0; i < ringSize; i++) {
             var state = orders[i];
-            var prev = orders[(i + ringSize - 1) % ringSize];
-            var next = orders[(i + 1) % ringSize];
+            var prevSplitB = orders[(i + ringSize - 1) % ringSize].splitB;
+            var nextIdx = (i + 1) % ringSize;
```
Comment (dong77) [inline contracts/LoopringProtocolImpl.sol]: please change prevSplitB and nextIdx to `uint`

Comment (dong77) [reply contracts/LoopringProtocolImpl.sol]: I can see why introduction of ` prevSplitB` may save some gas, but removing `var next` is supposed to increase gas....  can you try to revert removing `var next`?

**Comment by Brechtpd** [comment]
Did some additional changes that saves an additional 1000-1500 gas. I stripped availableAmountS out of OrderState so that I could have a local variable to cache uintArgsList[i] in assembleOrders. Let me know if you're not okay with that, though that also reduces memory use.

File: contracts/LoopringProtocolImpl.sol (hunk 1 above)
Comment (Brechtpd) [reply contracts/LoopringProtocolImpl.sol]: Seems like the 'next' problem solved itself with the latest updates on the master. :)

**Pull request merged**